### DISCUSSION TITLE
Unexpected page refresh on form submission with revalidatePath

### DISCUSSION THREAD

**Discussion by SlavenDj** [vercel/next.js#68144] [Help]
## Issue Description

I'm experiencing an unexpected page refresh when submitting a form for the first time after a page load. This only occurs when I call the `revalidatePath` function. Subsequent form submissions work normally. The issue seems to be isolated to the first submission after a page refresh.

## Code Overview

I'm using Next.js 14.2.4 with React 18.3.1. Here's a simplified version of my component:

```tsx
'use client';
import { handleDeleteFAQ } from '@/lib/actions/delete';
import { useFormState } from 'react-dom';

export default function FaqDeleteForm({ id }: { id: number }) {
  const [state, action] = useFormState(handleDeleteFAQ, { sent: false });

  return (
    <form action={action}>
      <input hidden type='text' value={id} readOnly name='id' />
      <Button type='submit' variant={'destructive'} size={'sm'}>
        <Trash className='size-4' />
      </Button>
    </form>
  );
}
```
The handleDeleteFAQ function:
```tsx
export const handleDeleteFAQ: handleDeleteactionType = async (prevState, formData) => {
  try {
    const id = +DeleteReqSchema.parse(Object.fromEntries(formData)).id;
    await deleteFAQ(id);
    revalidatePath('/dashboard');
    return { sent: true, id: id };
  } catch (error) {
    console.error(error);
    return { sent: true, error: 'error' };
  }
};
```

Questions

What might be causing this unexpected refresh on the first form submission?
How can I resolve the unhandled error issue despite having an error.tsx file?

Any insights or suggestions would be greatly appreciated. Thank you!

### Additional information

```js
Environment

Next.js: 14.2.4
React: 18.3.1
Node: 20.15.1
OS: macOS (Darwin Kernel Version 23.5.0)

Additional Context

I'm using Supabase for data fetching.
The issue only occurs when calling revalidatePath.
I have an error.tsx file, but I'm still seeing unhandled error messages.
```

**Comment by Juneezee** [comment]
Unfortunately, this seems to be a "feature" rather than a "bug".

This discussion https://github.com/vercel/next.js/discussions/61836 has a similar situation as yours. Hope it helps!

**Reply by SlavenDj** [reply]
Hi there, sorry for the late response. Unfortunately it doesn't not help at all and I have no idea what to do next. How could unexpected reload be a feature?
